### ISSUE TITLE
Problem with the Forms - Hydra iT

### ISSUE THREAD

**Issue by cancelaconcepts** [Orckestra/C1-CMS-Foundation#447]
Good morning.

I sent an email yesterday, which I got a reply to try and post the issue I have here. The quote is as it follows (for you to get an preview of the main issue):

_"Ticket #8777: Support Request - Hydra iT_

_Good afternoon,
Our company has a website running with it’s own domain. We use your software to manage it, and everything has been working smoothly… except the forms.
We have the main website of Hydra iT (our company, www.hydra.pt ), where the forms work. But we created a new website named www.flybusiness.pt, also based on your software. But, even if it has the same codes (razor functions, handlers), for some reason, the emails are not sent… even if we change to the emails of the current domain. We tried altering the code millions of times, but we weren’t successful (we have people related with Web Developing and they weren’t able to solve even changing the code). Also, the error that occurs it’s only a vague one: “error, Internal Server Error”. From what the specialists here can tell, the problem must be something related, not with the code, but with the email server or any other thing related with it, or maybe it can be related with the C1 CMS (could it be because the C1 CMS has a different version that affects the forms?)
By the way, the version of the C1 CMS Composite on www.hydra.pt (http://www.hydra.pt) is:
Composite C1 4.2 Update 1
Build no. 4.2.5287.17495
© 2017 Composite A/S,
and the www.flybusiness.pt (http://www.flybusiness.pt) is:
C1 CMS
Build no. 6.1.6325.31805
© 2017 Orckestra Inc
Having all of this in mind, could you help us in any way to solve this issue?
We would appreciate it!
Our cordial greetings,
Francisco Cancela"_

Which we got the response:
_"Hey Francisco,_

_probably this is the SMTP settings for your website - you should check the <smtp /> settings in your web.config._

_Please be aware that - since the software is free and open source - we cannot give private support for free as well. You can either use one of our public channels, like issues on GitHub or our Gitter chat channel, or get support for your website through our Business subscription. Thank you for your understanding._

_Marcus Wendt
C1 CMS Team"_

I also want to add more details that I hadn't mentioned before:

- What the form does is get some information from the input fields of the form and append them to a FormData. This function is definitely working when we tried to use a debugger. After this, an Ajax function is called in order to send this information through a handler we created, which has a format of ashx. . And here is where all the things go wrong: it looks like the Ajax function is not being read for some reason...  look like it is being skipped;
- About the response sent to us through the email... even by adding that bit of code, it still doesn't work (the www.hydra.pt website doesn't have that line of code by the way... but it still works!). Here is the code added by the way, which is inside the configuration tag, and it has also has the information related with the used domain (host and port):
<system.net> 
    <mailSettings> 
      <smtp deliveryMethod="Network"> 
        <network host="plesk02.host-services.com" port="465" defaultCredentials="true" /> /> 
      </smtp> 
    </mailSettings> 
  </system.net> 
(for some reason, I can't make the code I pasted here appear correctly!)

If you need any more information related with this, just tell me! 
I hope you can help us solve this issue.

Greetings,
Francisco Cancela
Hydra iT

**Comment by mawtex** [comment]
If you use F12 Developer Tools in the browser and look at the calls to the ashx, are there any errors there if you dive into the request/response details?

**Comment by cancelaconcepts** [comment]
Hello.

I did what you mentioned... and it was occurring the "Error 500 Internal Server" with the ashx file. It occured after pressing the form sending button. Even with this error, the data that was written on the input fields was being shown... and this also meant that the handler was being read (and in a sense, the ajax function was working).
But after this point, I was able to find out what the problem was.
It was related to this code which was used in the handler: _using LogNamespace;_. For some reason, the libraries related with this aren't present (that's what it was said at the error at least... when I tried to open directly the ashx file through the browser. To see this error, I had to disable the CustomError function written in the web.config). After deleting this line and the related code (which was unnecessary for it to function correctly), the handler started working! 

Either way, I want to thank you for the feedback! This issue can be considered "solved".


Greetings,
Francisco Cancela
Hydra iT
